src/Shape/Box.js

```diff
@@ -0,0 +1,267 @@
+    this._rectWidth = constant(50);
+    this._value = accessor("value");
+    this._whiskerMode = ["extent", "extent"];
+    this._x = constant(250);
+    this._y = constant(250);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Filters/manipulates the data array before binding each point to an SVG group.
+      @param {Array} [*data* = the data array to be filtered]
+      @private
+  */
+  _dataFilter(data) {
+    const boxes = this._box = nest().key(this._id).entries(data).map(d => {
+
+      d.data = merge(d.values);
+      const values = d.values.map(this._value);
+      values.sort((a, b) => a - b);
+      d.i = data.indexOf(d.values[0]);
+      d.x = this._x(d.data, d.i);
+      d.y = this._y(d.data, d.i);
+
+      d.first = quantile(values, 0.25);
+      d.median = quantile(values, 0.50);
+      d.third = quantile(values, 0.75);
+
+      d.width = this._rectWidth();
+      d.height = this._rectHeight(d);
+      
+      const mode = this._whiskerMode;
+
+      if (mode[0] === "tukey") d.top = d.first - (d.third - d.first) * 1.5;
+      else if (mode[0] === "extent") d.top = max(values);
+      else if (typeof mode[0] === "number") d.top = min([max(values), quantile(values, mode[0] / 100)]);
+
+      if (mode[1] === "tukey") d.bottom = d.third + (d.third - d.first) * 1.5;
+      else if (mode[1] === "extent") d.bottom = min(values);
+      else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
+
+      this._topWhiskerLength = d.top - d.third;
+      this._bottomWhiskerLength = d.first - d.bottom;
+
+      d.nested = true;
+      d.__d3plusShape__ = true;
+
+      return d;
+    });
+
+    boxes.key = d => d.key;
+    return boxes;
+
+  }
+
+  /**
+      @memberof Box
+      @desc Draws the Box.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const filteredData = this._dataFilter(this._data);
+
+    // Draw box.
+    new Rect()
+      .data(filteredData)
+      .x(filteredData[0].x)
+      .y(filteredData[0].y)
+      .config(this._rectConfig)
+      .select(elem("g.d3plus-box", {
+        parent: this._select
+      }).node())
+      .render();
+
+    // Compute coordinates for median line.
+    const boxTopY = this._y() - this._rectHeight(filteredData[0]) / 2;
+    const medianPoint1X = this._x() - this._rectWidth() / 2;
+    const medianPoint1Y = boxTopY + filteredData[0].third - filteredData[0].median;
+    const medianPoint2X = medianPoint1X + this._rectWidth();
+    const medianPoint2Y = medianPoint1Y;
+
+    const medianData = [
+      {x: medianPoint1X, y: medianPoint1Y},
+      {x: medianPoint2X, y: medianPoint2Y}
+    ];
+
+    // Draw median line inside the box.
+    new Line()
+      .data(medianData)
```

Comment:
let's use `Rect` for the median line, so we can just use `filteredData` for the `.data( )` method, and then you can set methods based on that data (example `.x(d => d.x - d.width / 2)`)

Reply:
You did not do what I asked you to do. We need to use `filterData` as the data array here so that users can set config methods based on the data they have passed. By creating a new `medianData` array with no reference to the user data, that becomes impossible. Please let me know if that is not clear.

To summarize:
* use `.data(filteredData)`
* when using the x, y, and rectWidth accessor functions, you NEED to pass the user data and index: `(d.data, d.i)`
* the the x, y, and rectWidth values already calculated as part of of each data point in filteredData (stored as `x`, `median`, and `width`), so you just need to set accessors on the `Rect` class to use them: `.y(d => d.median)`